### PR TITLE
chore: Only test from prometheus LTS n-1 and up to the latest stable

### LINKED ISSUE #5213: Reduce the number of versions in the end-to-end migration tests
I've checked on a recent CI run and [testPromVersionMigration](https://github.com/prometheus-operator/prometheus-operator/blob/1dfecab7c2706f3591d6bb6d5eb8f6b2518f1b45/test/e2e/prometheus_test.go#L824) takes more than 1500 seconds to complete. The reason is that the test starts from Prometheus v2.0.0 and deploys every Prometheus version up to the latest supported (v.2.40.5 today). Over time, the duration of the test can only increase up to the point where it will hit the global timeout. IMHO testing very old versions of Prometheus isn't terribly useful: nobody should be running a v2.0.0 Prometheus anymore and we should trim down the list of versions.

I suggest that we start the migration test from the before last [Prometheus TLS release](https://prometheus.io/docs/introduction/release-cycle/) up to the latest stable (right now there's only one TLS, v2.37 but as soon as the next LTS is out, it would be all versions >= v2.37.0). WDYT @PhilipGough @paulfantom?

### PR DESCRIPTION
**Pull request by xiu**
## Description

Before this commit, we were testing from v2.0.0 up until the latest. This made CI take an exponential amount of time, currently sitting at 1500 seconds. Additionally, it's unlikely that we have users still running versions as old as v2.0.0, the use for such test is limited.

Fixes #5213

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

```release-note
We reduced the span of supported versions: prometheus-operator will now support prometheus from version LTS n-1 to the latest supported version.
```

### CODE REVIEW

**Review by JoaoBraveCoding — COMMENTED** [review]
lgtm, but I'm curious, if we start doing this then shouldn't we also add somewhere in the README that prom-operator current version "supports Prometheus from the before last [Prometheus TLS release](https://prometheus.io/docs/introduction/release-cycle/) up to the latest stable"?

**Comment by xiu** [comment]
> lgtm, but I'm curious, if we start doing this then shouldn't we also add somewhere in the README that prom-operator current version "supports Prometheus from the before last [Prometheus TLS release](https://prometheus.io/docs/introduction/release-cycle/) up to the latest stable"?

I looked into it and we have this for compatibility: https://prometheus-operator.dev/docs/operator/compatibility/. I couldn't find a mention of it in the README, only pre-requisites.

**Comment by xiu** [comment]
@JoaoBraveCoding I added a note of this new rule in `./Documentation/compatibility.md`, let me know if you think it's sufficient.

**Review by JoaoBraveCoding — APPROVED** [review]
Yes, to me it looks good 👍 thanks for addressing it!

File: Documentation/compatibility.md [hunk 1]
```diff
@@ -27,75 +27,9 @@ v0.26.1
 
 ## Prometheus
 
-The versions of Prometheus compatible with the Prometheus Operator are:
+Prometheus Operator supports Prometheus from the before last [Prometheus TLS release](https://prometheus.io/docs/introduction/release-cycle/) up to the latest stable. The versions of Prometheus currently compatible with the Prometheus Operator are:
```
Comment (simonpasquier) [inline Documentation/compatibility.md:30]: the wording is a bit more "aggressive" than what I had in mind. IMHO we don't need to drop support for versions before LTS n-1. But it should be clear that the testing only covers TLS n-1 -> latest.
What about this?

```suggestion
Prometheus Operator supports all Prometheus versions >= v2.0.0. The operator's end-to-end tests verify that the operator can deploy the following Prometheus versoins:
```

File: pkg/operator/defaults.go [hunk 2]
```diff
@@ -45,74 +45,9 @@ var (
 	// configuration reloading
 	DefaultPrometheusConfigReloaderImage = "quay.io/prometheus-operator/prometheus-config-reloader:v" + version.Version
 
-	// PrometheusCompatibilityMatrix is a list of supported prometheus version
+	// PrometheusCompatibilityMatrix is a list of supported prometheus version.
+	// prometheus-operator supports versions ranging from LTS n-1 to the latest supported version
```
[the remark this answers is not in the document]
Comment (xiu) [reply pkg/operator/defaults.go:49]: With the following comment in mind, how about this:
```go
 	// PrometheusCompatibilityMatrix is a list of supported prometheus versions.
	// prometheus-operator end-to-end tests verify that the operator can deploy from LTS n-1 to the latest stable. 
	// This list should be updated every time a new LTS is released.
```

That way we make sure that contributors understand that the rule is for LTS n-1 to be the lower boundary.

[the remark this answers is not in the document]
Comment (xiu) [reply pkg/operator/defaults.go]: I've updated the MR with this for now

**Review by simonpasquier — APPROVED** [review]
thanks!

**Pull request merged**